client/index.js

```diff
@@ -0,0 +1,289 @@
+    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
+    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
+    return c > 3 && r && Object.defineProperty(target, key, r), r;
+};
+var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
+    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
+    return new (P || (P = Promise))(function (resolve, reject) {
+        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
+        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
+        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
+        step((generator = generator.apply(thisArg, _arguments || [])).next());
+    });
+};
+var Role;
+(function (Role) {
+    Role[Role["SUPERADMIN"] = 0] = "SUPERADMIN";
+    Role[Role["ADMIN"] = 1] = "ADMIN";
+    Role[Role["SUBSCRIBER"] = 2] = "SUBSCRIBER";
+})(Role || (Role = {}));
+;
+let crudObject; //array of objects to be displayed on frontend
+//DECORATOR FACTORY
+function FormatDate() {
+    return function (target, name, descriptor) {
+        const dateTime = document.getElementById("dateTime");
+        dateTime.innerHTML = new Date().toLocaleString();
+        setInterval(function () {
+            dateTime.innerHTML = new Date().toLocaleString();
+        }, 1000);
+    };
+}
+//MODEL CLASS FOR USER ENTRY
+class Crud {
+    constructor() {
+        this.items = [];
+    }
+    add(object) {
+        this.items.push(object);
+    }
+    update(id, object) {
+        editUser(id, object).then((message) => {
+            this.items = message.updatedMembers;
+            showTable();
+        }).catch(() => { alert("Unexpected Error Occured !"); });
+    }
+    delete(id, object) {
+        //make api call to delete data 
+        deleteUser(id).then((message) => {
+            //delete data on front end
+            this.items = message.members;
+            showTable();
+        }).catch(() => { alert("Unexpected Error Occured !"); });
+    }
+}
+__decorate([
+    FormatDate()
+], Crud.prototype, "add", null);
+//API CALLS ========================================================================================================================================
+function getUsers() {
+    return __awaiter(this, void 0, void 0, function* () {
+        let response = yield fetch('http://localhost:9000/api');
+        let users = yield response.json();
+        return users; // same as Promise.resolve(users)
+    });
+}
+function deleteUser(id) {
+    return __awaiter(this, void 0, void 0, function* () {
+        let response = yield fetch(`http://localhost:9000/api/${id}`, {
```

Comment:
api is not a valid keyword. CRUD apis should use the model name. Suppose, if you are writing crud of music. then it should be /musics